### ISSUE TITLE
Setup deployment of ontology via GitLab

### ISSUE THREAD

**Issue by jacobwindsor** [ebi-ait/dcp-ingest-central#285]
As a developer, I want to deploy the ontology service via an automated deployment pipeline.

Currently, all components except `ontology` are deployed via GitLab. `ontology` is still deployed using the Make commands. We could use GitLab to do this. This would involve setting up a new pipeline for the ontology service.

**Proposal**
1. Create a new Git repo that contains a `.gitlab-ci.yml`. We could also define a Gitlab pipeline for this in `ingest-kube-deployment`
2. Add a `ontology-version-history.txt` (or similar name) to the repo. This will contain the version numbers deployed from most recent to first. The CI pipeline will read the first line of this file and deploy that version.
3. Update the helm charts for `ontology` in `ingest-kube-deloyment` repo to allow deployments via GitLab
4. Commit changes to the `ontology-version-history.txt` and the CI pipeline will pick this up and deploy

**Comment by aaclan-ebi** [comment]
Is it possible to add the auto deployment of the validator after deployment of ontology service?

**Comment by tburdett** [comment]
Could we update the user story in this ticket to reflect the rationale (you could also consider that there's no "so that..." part to this story)?

I've asked in the past why we deploy our own version of the ontology rather than rely on the version in OLS and I'm not sure if we have a good (documented) answer to this question. Generally I really appreciate the approach to automate here, but at the same time it's crucial to be sure we all know why we need this in the first place, so as to ensure we're not incurring extra unnecessary maintenance costs or spending too much time fixing something we might be able remove.

**Comment by jacobwindsor** [comment]
@aaclan-ebi Yes! We can just trigger the deployment of that in the `.gitlab-ci.yml`. We will just have to override the deployment stages

**Comment by amnonkhen** [comment]
related to #800 which is a script that deploys the HCAO once it is released.
The missing part are:
- [ ] have it run by gitlab in response to some trigger, e.g. availability of a new HCAO release or tag in quay.io
- [ ] add some validation to the new release of HCA
